### PR TITLE
[netcore] Use mono-netcore to run real world apps using `dotnet run`

### PR DESCRIPTION
**Pull request by EgorBo**
Let's move the discussion here. 
So currently we have two .NET Core:
1) `mono-repo/.dotnet` - we use it to bootstrap and build stuff (the SDK version is specified in `global.json`) - if you open `netcore/Makefile` it's referred there as `$(DOTNET)` variable.
2) `mono-repo/netcore/.dotnet` - it's just a runtime + CoreFX libs (it doesn't contain SDK stuff, templates, msbuild, etc) - we use it only to run tests because this runtime is synchronized with corefx tests in `eng/Versions.prop` file which is updated by a bot and `netcore/shared` sources for our System.Private.CoreLib also depend on that exact runtime version (the sources also updated by the bot).

This PR introduces a rule to copy mono bits to the `mono-repo/.dotnet` folder to be able to run real world apps (at your own risk) using mono-netcore (so when you run `mono-root/.dotnet/.dotnet run -c Release` for an app - even MSBuild will use mono runtime to build that app). However in theory we need exact SDK for the runtime version we have but I have no idea where and how to get one.

BTW, currently MSBuild crashes when it's powered by mono on `$(DOTNET) build` command for a hello world.
```
monoeg_assertion_message
mono_domain_assembly_preload
invoke_assembly_preload_hook
...
at <unknown> <0xffffffff>
at System.Reflection.RuntimeAssembly:GetExportedTypes <0x000a2>
at AssemblyInfoToLoadedTypes:ScanAssemblyForPublicTypes <0x0008e>
```

### CODE REVIEW

**Comment by jaykrell** [comment]
Please just make sure mono and mono-netcore are both easy to use, from the command line, w/o long command lines and w/o lots of copy/paste. Both installed and uninstalled.
Even historical uninstalled mono could do a better job of finding its MONO_PATH, and now its System.Native.so. Hopefully they can both be better than mono-historical is now.

File: netcore/Makefile [hunk 1]
```diff
@@ -10,6 +10,10 @@ ASPNETCOREAPP_VERSION := 3.0.0-preview-18614-0151
 
 ROSLYN_VERSION := 3.3.0-beta2-19374-02
 
+# runtime version used by $(DOTNET) - local .net core sdk to bootstrap stuff
+# it doesn't match NETCOREAPP_VERSION
+BOOTSTRAP_RUNTIME = $(shell ls ../.dotnet/shared/Microsoft.NETCore.App)
```
[the remark this answers is not in the document]
Comment (filipnavara) [reply netcore/Makefile]: On clean checkout I ended up with `cp ../mono/mini/.libs/libmonosgen-2.0.dylib ../.dotnet/shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01 3.0.0-preview8-28379-02/libcoreclr.dylib`.

[the remark this answers is not in the document]
Comment (filipnavara) [reply netcore/Makefile]: `cat ../global.json | grep -m 1 "\"dotnet\"" | sed "s/.*\"dotnet\": \"\(.*\)\"/\\1/"`

[the remark this answers is not in the document]
Comment (EgorBo) [reply netcore/Makefile]: @filipnavara heh, I did the same initially but... global.json contains SDK version and we need Runtime version to access `libcoreclr.dylib` 🙂 
The only problem: `.dotnet` folder won't exist if system wide .NET Core matches our global.json (but there should be a switch somewhere to always download it to .dotnet)

[the remark this answers is not in the document]
Comment (filipnavara) [reply netcore/Makefile]: Are you telling me they don't match? *shrugs* Not that I am surprised.

**Comment by steveisok** [comment]
@monojenkins build Linux ARMv5

File: netcore/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (EgorBo) [reply netcore/Makefile]: yeah that is the most confusing thing for me every time in .NET versioning - these versions even look similar

File: eng/Versions.props [hunk 2]
```diff
@@ -9,6 +9,6 @@
     <!-- core-setup -->
     <MicrosoftNETCoreAppVersion>5.0.0-alpha1.19375.9</MicrosoftNETCoreAppVersion>
     <!-- roslyn -->
-    <MicrosoftNetCompilersVersion>3.3.0-beta2-19381-14</MicrosoftNetCompilersVersion>
```
Comment (EgorBo) [inline eng/Versions.props]: Turns out we are not ready for `3.3.0-beta2-19381-14` yet

**Pull request merged**